### ISSUE TITLE
cannot install on macos Ventura

### ISSUE THREAD

**Issue by whg517** [mitsuhiko/rye#184]
### Steps to Reproduce

### General operation (zsh), got error


```txt
 kevin@mac  /tmp  curl https://raw.githubusercontent.com/mitsuhiko/rye/main/scripts/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1713  100  1713    0     0   4813      0 --:--:-- --:--:-- --:--:--  4866
This script will automatically download and install rye (latest) for you.
######################################################################## 100.0%
Welcome to Rye!

This installer will install rye to /Users/kevin/.rye
This path can be changed by exporting the RYE_HOME environment variable.

Details:
  Rye Version: 0.1.0
  Platform: macos (aarch64)

Continue? [y/n] Error: Too many open files (os error 24)
```

Manual download, everything is fine.

```
 ✘ kevin@mac  /tmp  wget https://raw.githubusercontent.com/mitsuhiko/rye/main/scripts/install.sh
--2023-05-18 15:53:47--  https://raw.githubusercontent.com/mitsuhiko/rye/main/scripts/install.sh
正在解析主机 raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.108.133, 185.199.111.133, ...
正在连接 raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... 已连接。
已发出 HTTP 请求，正在等待回应... 200 OK
长度：1713 (1.7K) [text/plain]
正在保存至: “install.sh”

install.sh                                                               100%[================================================================================================================================================================================>]   1.67K  --.-KB/s  用时 0s      

2023-05-18 15:53:48 (5.48 MB/s) - 已保存 “install.sh” [1713/1713])

 kevin@mac  /tmp  chmod +x install.sh 
 kevin@mac  /tmp  ./install.sh 
This script will automatically download and install rye (latest) for you.
#=#=-  #       #                                                                                                                                                                                                                                               #=O#-     #        #                                                                                                                                                                                                                                           -#O=- #      #          #                                                                                                                                                                                                                                      -=O#-   #        #           #                                                                                                                                                                                                                                 -=O=-#      #        #            #                                                                                                                                                                                                                            -=O=-  #       #          #             #                                                                                                                                                                                                                      -=O=-     #        #           #              #                                                                                                                                                                   ######################################################################################################################################################################################################################################################### 100.0%
Welcome to Rye!

This installer will install rye to /Users/kevin/.rye
This path can be changed by exporting the RYE_HOME environment variable.

Details:
  Rye Version: 0.1.0
  Platform: macos (aarch64)

Continue? no
Installation cancelled!
```

### General operation (bash), got error too
```
 ✘ kevin@mac  /tmp  
 ✘ kevin@mac  /tmp  bash

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
bash-3.2$ 
bash-3.2$ ulimit -n 3000
bash-3.2$ 
bash-3.2$ curl https://raw.githubusercontent.com/mitsuhiko/rye/main/scripts/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:19 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:44 --:--:--     0^C
bash-3.2$ export http_proxy=http://127.0.0.1:1081
bash-3.2$ export https_proxy=http://127.0.0.1:1081
bash-3.2$ curl https://raw.githubusercontent.com/mitsuhiko/rye/main/scripts/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1713  100  1713    0     0    914      0  0:00:01  0:00:01 --:--:--   915
This script will automatically download and install rye (latest) for you.
######################################################################## 100.0%
Welcome to Rye!

This installer will install rye to /Users/kevin/.rye
This path can be changed by exporting the RYE_HOME environment variable.

Details:
  Rye Version: 0.1.0
  Platform: macos (aarch64)

Continue? [y/n] Error: Too many open files (os error 24)
bash-3.2$ 
bash-3.2$ 

```

### Expected Result

I used the command `curl https://raw.githubusercontent.com/mitsuhiko/rye/main/scripts/install.sh | bash
` to install rye as normal

### Actual Result

i can not install rye, and got error: `Continue? [y/n] Error: Too many open files (os error 24)`. 

### Version Info

I haven't installed rye properly yet so I don't think it has anything to do with the software version, but it must have something to do with the installation script. Because I tried to work with linux.

I can provide information about the current environment:

- os: `macos Ventura 13.3.1(a)`
- bash: `3.2` ?

### Stacktrace

_No response_

**Comment by mitsuhiko** [comment]
This is fascinating. I think this might be because the read key function might not like the redirected stdin.

**Comment by mitsuhiko** [comment]
Fixed in 0.1.1.
